jupyterlab_gcedetails/src/components/synchronized_sliders.tsx

```diff
@@ -0,0 +1,246 @@
+import * as React from 'react';
+import { stylesheet } from 'typestyle';
+import Grid from '@material-ui/core/Grid';
+import Slider from '@material-ui/core/Slider';
+import { TextInput } from 'gcp_jupyterlab_shared';
+import { HardwareConfiguration } from '../data';
+
+interface Props {
+  /* Assumes sorted list with memory and cpu set to 0 for first value */
+  values: HardwareConfiguration[];
+  onCpuChange: (value: number) => void;
+  onMemoryChange: (value: number) => void;
+}
+
+interface State {
+  memoryValue: number;
+  cpuValue: number;
+  memoryInputDisplayValue: string;
+  cpuInputDisplayValue: string;
+}
+
+export const STYLES = stylesheet({
+  container: {
+    minWidth: '450px',
+
+    /**
+     * Using overflow to deal with quirks of material-ui grid.
+     * See: https://material-ui.com/components/grid/#negative-margin
+     */
+    overflowX: 'hidden',
+    overflowY: 'hidden',
+  },
+  slider: {
+    marginLeft: '10px',
+  },
+  label: {
+    display: 'block',
+    minWidth: '100px',
+  },
+});
+
+export class SynchronizedSliders extends React.Component<Props, State> {
+  private memoryValues: number[];
+  private cpuValues: number[];
+
+  constructor(props: Props) {
+    super(props);
+
+    this.memoryValues = this.props.values.map(elem => elem.memory);
+    this.cpuValues = this.props.values.map(elem => elem.cpu);
+
+    this.state = {
+      memoryValue: this.memoryValues[0],
+      memoryInputDisplayValue: String(this.memoryValues[0]),
+      cpuValue: this.cpuValues[0],
+      cpuInputDisplayValue: String(this.cpuValues[0]),
+    };
+  }
+
+  private memoryFromCpu(cpu: number) {
+    const obj = this.props.values.find(elem => elem.cpu === cpu);
+    return obj.memory;
+  }
+
+  private cpuFromMemory(memory: number) {
+    const obj = this.props.values.find(elem => elem.memory === memory);
+    return obj.cpu;
+  }
+
+  /* Returns closest higher value to targetValue in possibleValues */
+  private closest(targetValue: number, possibleValues: number[]) {
+    return possibleValues.reduce((a, b) => {
+      return Math.abs(a - targetValue) < Math.abs(b - targetValue) ? a : b;
+    });
+  }
+
+  private onCpuChange(newCpu: number) {
+    this.setState({
+      cpuValue: newCpu,
+      cpuInputDisplayValue: String(newCpu),
+      memoryValue: this.memoryFromCpu(newCpu),
+      memoryInputDisplayValue: String(this.memoryFromCpu(newCpu)),
+    });
+
+    this.props.onCpuChange(newCpu);
+  }
+
+  private onMemoryChange(newMemory: number) {
+    this.setState({
+      cpuValue: this.cpuFromMemory(newMemory),
+      cpuInputDisplayValue: String(this.cpuFromMemory(newMemory)),
+      memoryValue: newMemory,
+      memoryInputDisplayValue: String(newMemory),
+    });
+
+    this.props.onMemoryChange(newMemory);
+  }
+
+  render() {
+    const {
+      cpuInputDisplayValue,
+      memoryInputDisplayValue,
+      cpuValue,
+      memoryValue,
+    } = this.state;
+
+    const handleCpuSliderChange = (event: any, newValue: number) => {
```

Comment:
As with the other component, I would suggest lifting these to private methods within the class.